### PR TITLE
Handle trailing slash on http urls

### LINKED ISSUE #10630: clear_download_cache is sensitive to trailing slash
### Description

@aarchiba , `clear_download_cache` appears to be sensitive to trailing slash. Is it a bug or feature? Will fixing this introduce other bugs for other variation of inputs?

This is a follow up issue of #10437.

### Expected behavior

If I tell it to clear `https://domain` whereas I cached the URL as `https://domain/`, there should be no difference.

### Actual behavior

Cached `https://domain/` isn't cleared.

### Steps to Reproduce

```python
>>> from astropy.utils.data import import_file_to_cache, get_cached_urls, clear_download_cache
>>> import_file_to_cache('https://example.com/', 'README.md')
'/path/to/.astropy/cache/download/url/182ccedb33a9e03fbf1079b209da1a31/contents'
>>> get_cached_urls()
['https://example.com/']
>>> clear_download_cache('https://example.com')  # No trailing slash
>>> get_cached_urls()
['https://example.com/']
>>> clear_download_cache('https://example.com/')
>>> get_cached_urls()                                                                          
[]
```

### PR DESCRIPTION
**Pull request by aarchiba**
### Description

This pull request is to ensure the cache mechanism treats `http://example.com` and `http://example.com/` equally.

Fixes #10630

EDIT: Follow up of  #10437

### CODE REVIEW

File: astropy/utils/tests/test_data.py [hunk 1]
```diff
@@ -2086,3 +2086,28 @@ def test_clear_download_cache_variants(temp_cache, valid_urls):
 def test_ftp_tls_auto(temp_cache):
     url = "ftp://anonymous:mail%40astropy.org@gdc.cddis.eosdis.nasa.gov/pub/products/iers/finals2000A.all"
     download_file(url)
+
+
+@pytest.mark.parametrize('base', ["http://exmaple.com", "https://example.com"])
```
Comment (pllim) [inline astropy/utils/tests/test_data.py:2091]: Is the typo intentional?
```suggestion
@pytest.mark.parametrize('base', ["http://example.com", "https://example.com"])
```

**Comment by pllim** [comment]
p.s. Ignore coverage failure as usual.

**Comment by aarchiba** [comment]
Er, not sure whether this wants a changelog entry? Otherwise fine.

**Comment by pllim** [comment]
Doesn't hurt to have the change log if you have already written it. Thanks!

**Pull request merged**